README.md

```diff
@@ -12,194 +12,129 @@ shared. These resources have a different lifecycle than resources in workspaces.
+      return "VirtualNetwork";
     }
-}
-```
-
-An inner class in the factory class is a good convention for implementing a Landing Zone Definition.
 
-```java
-public class FooLZFactory extends ArmClientsDefinitionFactory {
     ...
-
-    class FooLZDefinitionV1 extends LandingZoneDefinition {
-
-        protected FooLZDefinitionV1(
-                AzureResourceManager azureResourceManager, RelayManager relayManager) {
-            super(azureResourceManager, relayManager);
-        }
-
-        @Override
-        public Deployable definition(DefinitionContext definitionContext) {
-            var storage =
-                    azureResourceManager
-                            .storageAccounts()
-                            .define(definitionContext.resourceNameGenerator().nextName(20))
-                            .withRegion(Region.US_EAST2)
-                            .withExistingResourceGroup(definitionContext.resourceGroup());
-
-            var vNet =
-                    azureResourceManager
-                            .networks()
-                            .define(definitionContext.resourceNameGenerator().nextName(20))
-                            .withRegion(Region.US_EAST2)
-                            .withExistingResourceGroup(definitionContext.resourceGroup())
-                            .withAddressSpace("10.0.0.0/28")
-                            .withSubnet("compute", "10.0.0.0/29")
-                            .withSubnet("storage", "10.0.0.8/29");
-
-            return definitionContext
-                    .deployment()
-                    .withResourceWithPurpose(storage, ResourcePurpose.SHARED_RESOURCE)
-                    .withVNetWithPurpose(vNet, "compute", SubnetResourcePurpose.WORKSPACE_COMPUTE_SUBNET)
-                    .withVNetWithPurpose(vNet, "storage", SubnetResourcePurpose.WORKSPACE_STORAGE_SUBNET);
-        }
+        
+    @Override
+    public List<ResourceNameRequirements> getResourceNameRequirements() {
+      return List.of(
+              new ResourceNameRequirements(
+                      getResourceType(), ResourceNameGenerator.MAX_VNET_NAME_LENGTH));
     }
 ```
 
-Resources are defined using the standard Azure Java SDK but with the following constraints to consider:
+It is important that step's implementation should be idempotent. Please take a look at Stairway developer guide [here](https://github.com/DataBiosphere/stairway/blob/develop/FLIGHT_DEVELOPER_GUIDE.md).
 
-- The purpose of the resources must be indicated in the deployment.
-- Resources not included in the deployment won't be created.
-- The `create()` method in the resource definition must not be called.
-- The resource definition must have the required configuration for a creation before it can be added to the deployment.
+Resource creation is defined using the standard Azure Java SDK. Together with defining desired properties for a resource it is also important to assign specific tags to a resource based on requirements. 
```

Comment:
I don't understand what it means to "assign specific tags to a resource based on requirements". Is there an example you can point to?

Reply:
It is just to empathize that each resource should be tagged.

Reply:
I will remove 'based on requirements'. It's obvious that there should be some requirements for this.

Reply:
But what does it mean to tag a resource?

Reply:
@sergiygetlin I'm still confused about what it means to tag a resource.

Reply:
To set tag together with all other properties specific to a resource.
For instance,

```
var vm = ComputeManager.vm.define("myVM")
               .withIp()...
               ...
               .withTag() <- define tag for a resource.
               .create();
```